src/SW_Domain.c

```diff
@@ -0,0 +1,175 @@
+                             " Please select from \"xy\" and \"s\".",
+                             MyFileName, value);
+                }
+                strcpy(SW_Domain->DomainType, value);
+                break;
+            case 1: // Number of X slots
+                SW_Domain->nDimX = atoi(value);
+                break;
+            case 2: // Number of Y slots
+                SW_Domain->nDimY = atoi(value);
+                break;
+            case 3: // Number of S slots
+                SW_Domain->nDimS = atoi(value);
+                break;
+            #endif
+
+            case 4: // Start year
+                y = atoi(value);
+
+                if (y < 0) {
+                    CloseFile(&f, LogInfo);
+                    LogError(LogInfo, LOGFATAL,
+                             "%s: Negative start year (%d)", MyFileName, y);
+                }
+                SW_Domain->startyr = yearto4digit((TimeInt) y);
+                break;
+            case 5: // End year
+                y = atoi(value);
+
+                if (y < 0) {
+                    CloseFile(&f, LogInfo);
+                    LogError(LogInfo, LOGFATAL,
+                             "%s: Negative ending year (%d)", MyFileName, y);
+                }
+                SW_Domain->endyr = yearto4digit((TimeInt) y);
+                break;
+            case 6: // Start day of year
+                SW_Domain->startstart = atoi(value);
+                fstartdy = swTRUE;
+                break;
+            case 7: // End day of year
+                tempdoy = atoi(value);
+                SW_Domain->endend = (tempdoy == 365) ?
+                                Time_get_lastdoy_y(SW_Domain->endyr) : 365;
+                fenddy = swTRUE;
+                break;
+            case KEY_NOT_FOUND: // Unknown key
+                LogError(LogInfo, LOGWARN, "%s: Ignoring an unknown key, %s",
+                         MyFileName, key);
+                break;
+        }
+    }
+
+	if (SW_Domain->endyr < SW_Domain->startyr) {
+		CloseFile(&f, LogInfo);
+        LogError(LogInfo, LOGFATAL, "%s: Start Year > End Year", MyFileName);
+	}
+
+    // Check if start day of year was not found
+    if(!fstartdy) {
+        LogError(LogInfo, LOGWARN, "Domain.in: Missing Start Day - using 1\n");
+        SW_Domain->startstart = 1;
+    }
+
+    // Check if end day of year was not found
+	if (!fenddy) {
+		SW_Domain->endend = Time_get_lastdoy_y(SW_Domain->endyr);
+        LogError(LogInfo, LOGWARN,
+                "Domain.in: Missing End Day - using %d\n", SW_Domain->endend);
+	}
+
+    #ifdef SWNETCDF
+    // Check if the provided domain netCDF does not exist and needs to be created
+    if(!FileExists(PathInfo->InFiles_nc[DOMAIN_NC])) {
+        SW_NC_create_domain(SW_Domain, PathInfo->InFiles_nc[DOMAIN_NC],
+                            LogInfo);
+    }
+    #else
+    // Default domain type to "s" and number of sites to one
```

Comment:
Why do you overwrite `DomainType` and `nDimS` that just have been read from "domain.in" if `SWNETCDF` is not defined? This is against what we formulated in the plan.

Reply:
I'm not sure I see the problem here. The program will ignore any spacial domain information when SWNETCDF is not defined. Then it will default `DomainType` to "s" and `nDimS` to 1, not overwriting but setting the variables. Then if SWNETCDF is defined, we do read in spacial domain information from `domain.in` and will not be overwritten by that chunk of code.

The plan states
> New input variables (these default to “s” and  #s = 1 if not SWNETCDF)

Reply:
> New input variables (these default to “s” and #s = 1 if not SWNETCDF)

are provided as defaults via "domain.in"; they shouldn't be hard-coded